### PR TITLE
feat: add commands for exporting current and expected database schema

### PR DESCRIPTION
**Pull request by icewind1991**
Can be useful when investigating database related issue.

### CODE REVIEW

File: core/Command/Db/SchemaEncoder.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	}
+
+	private function encodeTable(Table $table, AbstractPlatform $platform): array {
+		$encoded = ['columns' => [], 'indexes' => []];
+		if ($table->getName() === 'oc_filecache_extended') {
+			$a = 1;
+		}
+		foreach ($table->getColumns() as $column) {
+			$data = $column->toArray();
+			$data['type'] = $column->getType()->getName();
```
[the remark this answers is not in the document]
Comment (come-nc) [reply core/Command/Db/SchemaEncoder.php:42]:
```
    /**
     * Gets the name of this type.
     *
     * @deprecated this method will be removed in Doctrine DBAL 4.0,
     *             use {@see TypeRegistry::lookupName()} instead.
     *
     * @return string
     */
    abstract public function getName();
```
This method is deprecated.

**Review by come-nc — COMMENTED** [review]
Looks good apart from deprecated method call.

**Review by artonge — APPROVED** [review]
Really good addition!

File: core/Command/Db/ExportSchema.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	}
+
+	protected function execute(InputInterface $input, OutputInterface $output): int {
+		$schema = $this->connection->createSchema();
+		$sql = $input->getOption('sql');
+		if ($sql) {
+			$output->writeln($schema->toSql($this->connection->getDatabasePlatform()));
+		} else {
+			$encoder = new SchemaEncoder();
+			$this->writeArrayInOutputFormat($input, $output, $encoder->encodeSchema($schema, $this->connection->getDatabasePlatform()));
```
Comment (come-nc) [inline core/Command/Db/ExportSchema.php:39]: We should add support for iterables in there to some day be able to output while computing when not using json output.

**Review by come-nc — APPROVED** [review]

**Pull request merged**